src/components/layouts/SplitPane.tsx

```diff
@@ -0,0 +1,260 @@
+	overflow: 'hidden',
+};
+
+const preview = {
+	display: 'grid',
+	gap: '3px',
+	height: '100%',
+	width: '100%',
+	position: 'absolute',
+	overflow: 'hidden',
+	top: '0',
+	left: '0',
+	zIndex: '100',
+};
+
+const previewPart = {
+	height: '100%',
+	width: '100%',
+	position: 'relative',
+	overflow: 'hidden',
+	background: 'rgba(138, 138, 138, 0.5)',
+	borderRadius: '20px',
+};
+
+const splitter = {
+	gridArea: 'splitter',
+	position: 'relative',
+	transition: '250ms',
+	borderRadius: '15px',
+	'&:hover': {
+		background: 'rgb(0, 0, 0, 0.1)',
+	},
+	'&:active': {
+		background: 'rgb(0, 0, 0, 0.2)',
+	},
+};
+
+const useStyles = createUseStyles<string, StylesProps>({
+	verticalSplitPane: props => ({
+		gridTemplateAreas: `
+             "first"
+             "splitter"
+             "second"
+         `,
+		gridTemplateRows: props.columns,
+		gridTemplateColumns: props.rows,
+		...pane,
+	}),
+	horizontalSplitPane: props => ({
+		gridTemplateAreas: `
+             "first splitter second"
+         `,
+		gridTemplateRows: props.rows,
+		gridTemplateColumns: props.columns,
+		...pane,
+	}),
+	verticalSplitPreview: props => ({
+		gridTemplateAreas: `
+             "first"
+             "splitter"
+             "second"
+         `,
+		gridTemplateRows: props.columnsPreview,
+		gridTemplateColumns: props.rowsPreview,
+		cursor: 'row-resize',
+		...preview,
+	}),
+	horizontalSplitPreview: props => ({
+		gridTemplateAreas: `
+             "first splitter second"
+         `,
+		gridTemplateRows: props.rowsPreview,
+		gridTemplateColumns: props.columnsPreview,
+		cursor: 'col-resize',
+		...preview,
+	}),
+	splitPreviewFirst: {
+		gridArea: 'first',
+		...previewPart,
+	},
+	splitPreviewSecond: {
+		gridArea: 'second',
+		...previewPart,
+	},
+	splitPaneFirst: {
+		gridArea: 'first',
+		...panePart,
+	},
+	splitPaneSecond: {
+		gridArea: 'second',
+		...panePart,
+	},
+	verticalSplitter: {
+		cursor: 'row-resize',
+		'&:before, &:after': {
+			content: '""',
+			position: 'absolute',
+			left: 'calc(50% - 15px)',
+			width: '30px',
+			height: '1px',
+			display: 'block',
+			background: 'rgb(0, 0, 0, 0.5)',
+		},
+		'&:before': {
+			top: '45%',
+		},
+		'&:after': {
+			top: '55%',
+		},
+		...splitter,
+	},
+	horizontalSplitter: {
+		cursor: 'col-resize',
+		'&:before, &:after': {
+			content: '""',
+			position: 'absolute',
+			top: 'calc(50% - 15px)',
+			width: '1px',
+			height: '30px',
+			display: 'block',
+			background: 'rgb(0, 0, 0, 0.5)',
+		},
+		'&:before': {
+			left: '45%',
+		},
+		'&:after': {
+			left: '55%',
+		},
+		...splitter,
+	},
+});
+
+export interface Props {
+	first: React.ReactNode;
+	second: React.ReactNode;
+	isVertical?: boolean;
+	firstMinSize?: number;
+	secondMinSize?: number;
+}
+
+function SplitPane({
+	first,
+	second,
+	isVertical = false,
+	firstMinSize = 0,
+	secondMinSize = 0,
+}: Props) {
+	const [offset, setOffset] = useState(50);
+	const [offsetPreview, setOffsetPreview] = useState(50);
```

Comment:
This is not very good solution. In this case, we will call the render component every time state changes.
We need to set the size of the element directly through the reference.